### PR TITLE
THRIFT-5648: add uuid support in rust

### PR DESCRIPTION
**Pull request by Jimexist**
- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

File: lib/rs/Cargo.toml [hunk 1]
```diff
@@ -15,10 +15,14 @@ keywords = ["thrift"]
 [dependencies]
 byteorder = "1.3"
 integer-encoding = "3.0.3"
+uuid = { version = "1.2.0" }
```
Comment (Jimexist) [inline lib/rs/Cargo.toml:18]:
```suggestion
uuid = "1"
```

**Comment by Jimexist** [comment]
@allengeorge per discussion in #2700, @fishy suggested that we do not bring any 3rd dependencies (in go library). here we are facing the same issue, do you believe we shall remove the uuid dependency in rust?

**Comment by fishy** [comment]
> @allengeorge per discussion in #2700, @fishy suggested that we do not bring any 3rd dependencies (in go library). here we are facing the same issue, do you believe we shall remove the uuid dependency in rust?

go library is slightly different in that it already has zero third party dependency right now so there's a big incentive to keep it that way, which does not apply to rust library (which I _believe_ already have third party dependencies).

**Comment by allengeorge** [comment]
I'm ok with adding a dependency (we shouldn't add them without thought, of course). The Rust UUID crate is dual-licensed ASL 2.0/MIT, so that's not an issue.

**Comment by Jimexist** [comment]
> I'm ok with adding a dependency (we shouldn't add them without thought, of course). The Rust UUID crate is dual-licensed ASL 2.0/MIT, so that's not an issue.
> 
> We can always reimplement the UUID portions necessary and remove the dependency over time.

thanks for the context. @allengeorge in this case do you think this pr can be merged as is?

**Comment by Jimexist** [comment]
hi @allengeorge and @Jens-G please take a look at this, thanks1

**Comment by my-vegetable-has-exploded** [comment]
I'd like to know why this pr was closed. thanks

**Comment by Jimexist** [comment]
> I'd like to know why this pr was closed. thanks

hi, simple answer - i was not able to get approvers for this.

**Comment by my-vegetable-has-exploded** [comment]
> > I'd like to know why this pr was closed. thanks
> 
> hi, simple answer - i was not able to get approvers for this.

what a pity！

**Comment by Jens-G** [comment]
> hi, simple answer - i was not able to get approvers for this.

For starters, I am not a Rust guy, and @allengeorge already chimed in. That's why I removed the review request. 

Next, we have plenty of pending PRs, which is a whole topic on itself. Getting no approvers applies to most of them, but the real question is why? IMHO this is not a indication of a quality issue, but a quantity issue. If we had enough active committers taking care, we would be able to give better responses than "there just was nobody around to take care".

**Comment by Jens-G** [comment]
I just skimmed over the changes and what could be better is the fact, that the compiler part also has a lot of refactorings in it. It's a bit hard to see what the real changes are for uuid and what is just shuffling comments around etc. I'd say we give it another chance. Shall we?

**Pull request merged**